src/Orleans/Core/GrainAttributes.cs

```diff
@@ -346,18 +347,26 @@ public LogConsistencyProviderAttribute()
                 ProviderName = Runtime.Constants.DEFAULT_LOG_CONSISTENCY_PROVIDER_NAME;
             }
         }
-
     }
 
     [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
     public sealed class ImplicitStreamSubscriptionAttribute : Attribute
```

Comment:
@jason-bragg I've added another ctor to `ImplicitStreamSubscriptionAttribute` and removed `sealed` to allow this kind of extensibility. I didn't go with the abstract base class you suggested, because 1) attributes cannot be generic thus you cannot put generic restrictions on the predicate type like `where TPredicate : IStreamNamespacePredicate, new()` 2) having a non-abstract attribute with a ctor accepting the predicate type seems to be useful on its own (see `FilteredImplicitSubscriptionGrain` in tests) so that you don't have to implement your own attribute just to pass the predicate type. 

So in the end, it looked like there was no extra benefit of that base attribute type.